src/meta/src/rpc/ddl_controller.rs

```diff
@@ -304,11 +304,16 @@ impl CreatingStreamingJobPermit {
                     }
                     Ordering::Equal => continue,
                     Ordering::Greater => {
-                        semaphore_clone
-                            .acquire_many((permits - new_permits) as u32)
-                            .await
-                            .unwrap()
-                            .forget();
+                        let to_release = permits - new_permits;
+                        let reduced = semaphore_clone.forget_permits(to_release);
+                        // TODO: implement dynamic semaphore with limits by ourself.
+                        if reduced != to_release {
```

Comment:
Shall we assign `permits` to the **actual** new permits here?

Reply:
No, some permits may have been acquired by some creating jobs and not returned yet.

Reply:
I mean the `permits` variable indicating the total permits, instead of the number of available permits inside the semaphore. 👀

Reply:
Oh yes, that's what I'm gonna to do and why I left a `TODO` here. I will fix it together with uncovered scenarios in recovered background jobs.